src/common/error.ts

```diff
@@ -0,0 +1,52 @@
+import { FastifyReply } from "fastify";
+
+export enum ErrorCodes {
+    ValidationError,
+    AuthError,
+    MessageError,
+    DataError,
+    ConflictError,
+    PermissionError
```

Comment:
style: trailing comma on multiline objects for better diffs
```suggestion
    PermissionError,
```